src/tile.js

```diff
@@ -125,7 +132,7 @@ export default class Tile {
                         // Parse style
                         rule.name = rule.name || StyleParser.defaults.style.name;
                         style = styles[rule.name];
-                        feature_style = style.parseFeature(feature, rule, tile, context);
+                        feature_style = style.parseFeature(feature, rule, source, context);
```

Comment:
I was wondering why `source` was being substituted for `tile` here... then I went and looked at `parseFeature` and it seems like neither is needed anymore (the tile is spit out in an error message, but that could be omitted, or moved here).